docs/design_proposals/log-selectors.md

```diff
@@ -0,0 +1,102 @@
+# Title
+
+* Author(s): Corneius Weig (@corneliusweig)
+* Design Shepherd: Priya Wadhwa (@priyawadhwa)
+* Date: 21/04/2019
+* Status: [Reviewed/Cancelled/Under implementation/Complete]
+
+## Background
+
+Skaffold offers the possibility to watch pod logs for the `dev` and `run` subcommands.
+So far, the pods to watch are determined by looking for known artifact images in the containers.
+This is restricting for two reasons:
+1. It does not allow to add additional pods to log aggregator which are not deployed by Skaffold (#666).
+2. It does not allow to exclude uninteresting pods deployed by Skaffold. For example, when working with istio, all the irrelevant istio logs spam the log (#588) 
+
+## Design
+I suggest to change the selection of pods for the log aggregator based on labels on this pod.
+
+For that, we need to change the definition of the `SkaffoldRunner` as follows:
+```go
+type SkaffoldRunner struct {
+	build.Builder
+	deploy.Deployer
+	test.Tester
+	tag.Tagger
+	sync.Syncer
+	watch.Watcher
+
+	cache             *cache.Cache
+	opts              *config.SkaffoldOptions
+	labellers         []deploy.Labeller
+	builds            []build.Artifact
+	hasBuilt          bool
+	hasDeployed       bool
+	
+	// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+	// OLD
+	// imageList         *kubernetes.ImageList
+	// NEW
+	podSelector       kubernetes.PodSelector
+	// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+	
+	RPCServerShutdown func() error
+}
+```
+This change does not affect the public API.
+
+To support the new use-cases of altering the pod selector, I propose a new top-level config section in the pipeline config:
+```yaml
+log:
+  # a list of deployment.spec.selector items, rules are OR'ed
+  - matchExpressions: # in one item, rules are AND'ed
+      - key: XY
+        values: ['val1', 'val2']
+        operator: In NotIn Exists DoesNotExist
+    # or
+    matchLabels: # in one item, rules are AND'ed
+      tail: "true"
+```
+This structure is based on the standard selector spec for k8s deployments.
```

Comment:
Could we add this [link](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors) here in case anyone is curious? thanks!

Reply:
good idea!